### PR TITLE
Closes #1712 deprecating and removing functions

### LINKED ISSUE #1681: General Issue: Deprecate `derive_var_disposition_status()`
### Background Information

`derive_var_disposition_status()` should be deprecated in favor of `derive_var_merged_cat()`

### Definition of Done

- [x] `derive_var_disposition_status()` is deprecated
- [x] vignettes are updated
- [x] templates are updated

### LINKED ISSUE #1682: General Issue: Deprecate `derive_var_worst_flag()`
### Background Information

`derive_var_worst_flag()` should be deprecated in favor of `slice_derivation()`/`derive_var_extreme_flag()`.

### Definition of Done

- [ ] `derive_var_worst_flag()` is deprecated
- [ ] vignettes are updated
- [ ] templates are updated

### LINKED ISSUE #1683: General Issue: Deprecate `derive_var_disposition_reason()`
### Background Information

`derive_var_disposition_reason()` should be deprecated in favor of `restrict_derivation()`/`derive_vars_merged()`.

### Definition of Done

- [ ] `derive_var_disposition_reason()` is deprecated
- [ ] vignettes are updated
- [ ] templates are updated

### LINKED ISSUE #1712: General Issue: Review vignettes and templates for deprecated functions
### Background Information

Go through the templates and vignettes to ensure that all of the functions and/or parameters that were deprecated are following the deprecation process outlined in the programming strategy (screenshot below)
image

### Definition of Done

- [x] Functions and/or parameters with **deprecated_stop()** from previous release are no longer in any vignettes or templates
- [x] Vignette sections with the deprecated function and/or parameters are updated with their respective replacements

Actions required for specific functions
- [x] Following functions need to **throw an error** for **release Q1-2023** admiral version 0.10.0 because they were deprecated in version 0.8.0 with phase 1, but haven't gone through Phase 2 which is to throw an error **OR** the deprecation process started in admiral version 0.9.0
`derive_vars_merged_dt()`
`derive_vars_merged_dtm()`
`derive_var_agegr_ema()`
`derive_var_agegr_fda()`
`derive_param_first_event()`
`derive_derived_param()`

- [x] Remove the following parameters
[smq_select()](https://pharmaverse.github.io/admiral/reference/smq_select.html) and [sdg_select()](https://pharmaverse.github.io/admiral/reference/sdg_select.html) function
`meddra_version`, `whodd_version`, `get_smq_fun` and `get_sdg_fun` from the [create_query_data()](https://pharmaverse.github.io/admiral/reference/create_query_data.html) function
`date_imputation`, `time_imputation` and `preserve` from `data_source()`

- [x] Following programs and their associated test in /test-deprecations.R need to be **removed** since they were deprecated with **Phase 2** in admiral version 0.8.0 or early
[`R/derive_var_ady.R` ](https://github.com/pharmaverse/admiral/blob/main/R/derive_var_ady.R)
[`R/derive_var_aendy.R`](https://github.com/pharmaverse/admiral/blob/main/R/derive_var_aendy.R)
[`R/derive_var_astdy.R`](https://github.com/pharmaverse/admiral/blob/main/R/derive_var_astdy.R)
[`R/derive_var_atirel.R`](https://github.com/pharmaverse/admiral/blob/main/R/derive_var_atirel.R)

### PR DESCRIPTION
**Pull request by sadchla-codes**
…emoving several functions. See issue #1712 for a list

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -48,6 +48,31 @@ added for subjects who have both an event or censoring and an observation in
 `HYPERURICEMIA` as grade criteria based on `ANRHI` only.  This metadata holds criteria for lab grading
 based on [Common Terminology Criteria for Adverse Events (CTCAE) v5.0](https://ctep.cancer.gov/protocoldevelopment/electronic_applications/ctc.htm) (#1650)
 
+- The following functions has been deprecated from previous {admiral} versions using the next phase of the deprecation process: (#1712)
```
Comment (bms63) [inline NEWS.md:51]:
```suggestion
- The following functions have been deprecated from previous `{admiral}` versions using the next phase of the deprecation process: (#1712)
```

**Review by bms63 — APPROVED** [review]
Just a couple of minor news comments.  Many thanks for taking this on @sadchla-codes.!!

Can you also make a new placeholder issue for admiral 0.11.0 with Q2 release label so we can keep line of sight.

File: R/derive_merged.R [hunk 2]
```diff
@@ -370,46 +370,11 @@ derive_vars_merged_dt <- function(dataset,
...
                                   duplicate_msg = NULL) {
-  assert_vars(by_vars)
-  dtc <- assert_symbol(enexpr(dtc))
-  filter_add <- assert_filter_cond(enexpr(filter_add), optional = TRUE)
-  assert_data_frame(dataset_add, required_vars = expr_c(by_vars, dtc))
-
-  deprecate_warn(
-    "0.8.0",
+  deprecate_stop(
+    "0.10.0",
```
Comment (bundfussr) [inline R/derive_merged.R:374]: Should we change the version number?
The message is now
```
`derive_vars_merged_dt()` was deprecated in admiral 0.10.0 and is now defunct.
```
Shouldn't it be
```
`derive_vars_merged_dt()` was deprecated in admiral 0.8.0 and is now defunct.
```
?
@sadchla-codes , @bms63 , what do you think?

Comment (sadchla-codes) [reply R/derive_merged.R:374]: @bundfussr it should have been 0.10.0? i'll make the updates

File: NEWS.md [hunk 3]
```diff
@@ -48,6 +48,31 @@ added for subjects who have both an event or censoring and an observation in
...
+
+  - `derive_var_ady()`
+  - `derive_var_aendy()`
+  - `derive_var_astdy()`
+  - `derive_var_atirel()`
+  - `derive_vars_suppqual()`
+  - `smq_select()`
+  - `sdg_select()`
+
+- The following parameters, which were deprecated in previous {admiral} versions, has been removed (#1712):
```
Comment (zdz2101) [inline NEWS.md:70]:
```suggestion
- The following parameters, which were deprecated in previous `{admiral}` versions, have been removed (#1712):
```

File: R/derive_merged.R (hunk 2 above)
Comment (bms63) [reply R/derive_merged.R:374]: `derive_vars_merged_dt()` was deprecated in admiral 0.8.0 and is now defunct. this look correct to me

Comment (sadchla-codes) [reply R/derive_merged.R:374]: making the updates now!

File: R/derive_vars_aage.R [hunk 4]
```diff
@@ -231,66 +232,23 @@ NULL
...
-      )
-    ) %>%
-    select(-temp_age)
-
-  if (anyNA(dplyr::pull(out, !!new_var))) {
-    out <- mutate(out, !!new_var := addNA(!!new_var))
-  }
-  out
+  deprecate_stop("0.10.0", "derive_var_agegr_fda()",
+    details = "Please create a user defined function instead."
```
Comment (zdz2101) [inline R/derive_vars_aage.R:240]: Perhaps include this statement in the roxygen documentation too?
image

File: NEWS.md [hunk 5]
```diff
@@ -48,6 +48,31 @@ added for subjects who have both an event or censoring and an observation in
...
+
+  - `derive_var_ady()`
+  - `derive_var_aendy()`
+  - `derive_var_astdy()`
+  - `derive_var_atirel()`
+  - `derive_vars_suppqual()`
+  - `smq_select()`
+  - `sdg_select()`
+
+- The following parameters, which were deprecated in previous {admiral} versions, have been removed (#1712):
```
Comment (zdz2101) [inline NEWS.md:70]: Just to match above
```suggestion
- The following parameters, which were deprecated in previous `{admiral}` versions, have been removed (#1712):
```

**Review by zdz2101 — APPROVED** [review]

**Review by bundfussr — APPROVED** [review]

**Comment by bms63** [comment]
I think these actions are failing on `insightenginnerring` end.

**Comment by sadchla-codes** [comment]
> I think these actions are failing on `insightenginnerring` end.

I'm not really familiar with that one, is this on github end?

**Comment by galachad** [comment]
The new release of stage dependencies action break the CI stage dependency action (https://github.com/insightsengineering/staged-dependencies-action/commit/a2186a876c2f3cbe756ed28d347de252b2dc9996). @cicdguy

**Comment by galachad** [comment]
Hi @cicdguy it's database corrupt issue. We should use other package to install remotes as it corrupt database. For local development, session restart is enough. It's happens as `remotes::install_github("r-lib/remotes@main")` load remotes namespace and overwrite it in the same moment.

Maybe it's possible to use, direct installation from URL [`tar.gz`](https://github.com/r-lib/remotes/archive/refs/heads/main.tar.gz)?

**Review by zdz2101 — APPROVED** [review]

**Review by bms63 — APPROVED** [review]
Whooo!!!!  Thanks for seeing this through!!  Have a good weekend!!

**Pull request merged**